src/util/serialize.h

```diff
@@ -326,6 +354,25 @@ inline void writeF32(u8 *data, f32 i)
 	throw SerializationError("writeF32: Unreachable code");
 }
 
+inline void writeF32(u8 *data, double i)
+{
+	writeF32(data, (f32)i);
+}
+
+inline void writeD1000(u8 *data, double i)
+{
+	// The limits are relevant to the 32 bit destination, not the float precision.
+	assert(i >= F1000_MIN && i <= F1000_MAX);
+	writeS32(data, i * FIXEDPOINT_FACTOR);
+}
+
+inline void writeD(u8 *data, double i)
+{
+	f32 base = (f32)i;
+	writeF32(&data[0], base);
```

Comment:
should have a short comment describing what this does and why

Reply:
I added comments. I am not sure if they pass as "short," but I think they are clear.